### PR TITLE
Remove `impl Foo for .. {}` in favor `auto trait Foo {}`

### PR DESCRIPTION
**Pull request by petrochenkov**
Rebase of https://github.com/rust-lang/rust/pull/46480 with restored parsing support.

### CODE REVIEW

**Comment by petrochenkov** [comment]
https://github.com/rust-lang/rust/pull/46480 is an old PR and should be at the top of the queue already.
@bors p=1

File: src/libsyntax/parse/parser.rs [hunk 1]
```diff
@@ -5421,48 +5421,35 @@ impl<'a> Parser<'a> {
...
-            self.expect(&token::OpenDelim(token::Brace))?;
-            self.expect(&token::CloseDelim(token::Brace))?;
-            Ok((keywords::Invalid.ident(),
-             ItemKind::AutoImpl(unsafety, opt_trait.unwrap()), None))
-        } else {
-            if opt_trait.is_some() {
-                ty = self.parse_ty()?;
+        if opt_trait.is_some() {
+            ty = if self.eat(&token::DotDot) {
+                P(Ty { node: TyKind::Err, span: self.prev_span, id: ast::DUMMY_NODE_ID })
```
Comment (eddyb) [inline src/libsyntax/parse/parser.rs]: Wait, does this actually trigger any errors, at all?

**Comment by petrochenkov** [comment]
Appveyor timeout :( https://github.com/rust-lang/rust/issues/46903
@bors retry

**Pull request merged**